### DISCUSSION TITLE
Can't speak with a mic

### DISCUSSION THREAD

**Discussion by PadowYT2** [truckersmp-cli/truckersmp-cli#311] [Q&A] [answered]
So while holding `x` I should speak, but I don't have anything popping out on the left top corner, as I've searched on TMP Discord,
"Make sure nothing else is on `x`, either reinstall TMP", as I just recently installed, why do I need to do that also I checked keybinds. Anyways any other explanations?

**Comment by Lucki** [comment, marked as answer]
I've never tried to use the microphone but here's how it should work:

* Make sure your microphone is selected as default input.
  You can check and change this with `pavucontrol`.
* Manually override the input of TruckersMP. This might be tricky as the program has to actively capture which might not be the case when using a push to talk key.
  * This can also be done in `pavucontrol`.
  * If you're using Pipewire you can use something like `qpwgraph` to graphically route your microphone to the TruckersMP input.

**Reply by PadowYT2** [reply]
I don't have `pavucontrol`, I use `pulseaudio` for all other games, so I'm supposed to have it?

**Reply by Lucki** [reply]
`pavucontrol` is just a graphical frontend for configuring pulseaudio. You don't have to use it - there are alternatives which you can use like some desktop environments have some options included - or even do it all manually using configuration files and `pactl`. For `pavucontrol` I know the setting is available.

You should check those mentioned settings out - regardless which program you decide to use.
